nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-shared/src/main/java/org/apache/nifi/kafka/shared/validation/KafkaClientCustomValidationFunction.java

```diff
@@ -233,6 +235,21 @@ private void validateUsernamePassword(final ValidationContext validationContext,
         }
     }
 
+    private void validateAWSIAMMechanism(final ValidationContext validationContext, final Collection<ValidationResult> results) {
+        final String saslMechanism = validationContext.getProperty(SASL_MECHANISM).getValue();
+
+        if (SaslMechanism.AWS_MSK_IAM.getValue().equals(saslMechanism) && !StandardKafkaPropertyProvider.isAwsMskIamCallbackHandlerFound()) {
```

Comment:
Minor: `SaslMechanism.getSaslMechanism()` could be used instead of the String comparison.
```suggestion
        final SaslMechanism saslMechanism = SaslMechanism.getSaslMechanism(validationContext.getProperty(SASL_MECHANISM).getValue());

        if (SaslMechanism.AWS_MSK_IAM == saslMechanism && !StandardKafkaPropertyProvider.isAwsMskIamCallbackHandlerFound()) {
```